### PR TITLE
Refactor/split netcode from clientmgr

### PR DESCRIPTION
**Pull request by dylanh724**
## About

Legacy scripts and evolving purposes ended up making HathoraClientMgr bound to NetworkManager calls. These should be completely separate, where actions like **Start/Stop Server/Client** and events like **OnClientStarted/OnClientStopped** should go to their own script. 

In addition, the HathoraClientMgr shouldn't be abstract, anymore, once separated.

## Changes

* Renamed **HathoraServerMgrBase** to **HathoraServerMgr**.
  * But kept the protected virtual members, ready to be inherited, should you want to.
* Renamed **HathoraClientMgrBase** to **HathoraClientMgr**.
  * But kept the protected virtual members, ready to be inherited, should you want to.
* Moved NetworkManager funcs/events to `NetworkMgrStateTracker`.
  * With protected virtual members, ready to be inherited, should you want to.
* Renamed **HathoraFishnetClientMgr** to **FishnetStateMgr**.
* Renamed **HathoraMirrorClientMgr** to **MirrorStateMgr**.
* Added additional docs/comments
* Carefully went through both FishnetStateMgr + MirrorStateMgr to ensure an almost 1:1 parity for clarity.
* Removed **HathoraFishnetServerMgr** demo script since we can just use **HathoraServerMgr** to accomplish the same thing (a Mirror variant already !existed).
* Child `Singleton` props that also have a base `Singleton` now added the `new` keyword to override the base one when active.

# Tests
* Tested in FishNet UDP
  * SDK Demo
  * Hello World Demo
* Tested in Mirror UDP
  * SDK Demo
  * Hello World Demo

## TL;DR

* Split away NetworkManager logic from HathoraClientMgrs to their own StateMgrs, serializing to NetworkManager instead of HathoraManager.
* Removed abstraction requirements of the Hathora managers

### CODE REVIEW

**Comment by dylanh724** [comment]
# Passing Tests

* Tested in FishNet UDP
  * SDK Demo
  * Hello World Demo
* Tested in Mirror UDP
  * SDK Demo
  * Hello World Demo
